File: LeetCode/cpp/970.cpp
class Solution {
public:
    vector<int> powerfulIntegers(int x, int y, int bound) {
        set<int> s;
        vector<int> ret;
        int i = 1, j = 1;
        while (i + j <= bound) {
            while (i + j <= bound) {
                s.insert(i + j);
                if (y == 1)
                    break;
                j *= y;
            }
            j = 1;
            if (x == 1)
                break;
            i *= x;
        }
        for (auto m:s)
            ret.push_back(m);
        return ret;
    }
};

File: LeetCode/python3/970.py
class Solution:
    def powerfulIntegers(self, x, y, bound):
        """
        :type x: int
        :type y: int
        :type bound: int
        :rtype: List[int]
        """
        i, j = 0, 0
        ret = set()
        while x ** i < bound:
            res = x**i + y**j
            while res <= bound:
                ret.add(res)
                if y == 1:
                    break
                j += 1
                res = x**i + y**j
            if x == 1:
                break
            j = 0
            i += 1
        return list(ret)

